b2share/modules/schemas/cli.py

```diff
@@ -154,124 +153,225 @@ def block_schema_edit(verbose, name, community, deprecated, block_schema_id):
         if comm:
             data['community'] = comm.id
         else:
-            click.secho("""Community not changed : no community exists with 
+            click.secho("""Community not changed : no community exists with
             name or id: %s""" % community)
     if deprecated:
-        if not type(deprecated)==bool:
+        if type(deprecated) != bool:
             raise click.BadParameter("""Deprecated should be True or False
              starting with a capital""")
         data['deprecated'] = deprecated
     block_schema.update(data)
     db.session.commit()
-        
+
+
 @schemas.command()
 @with_appcontext
-@click.option('-v','--verbose', is_flag=True, default=False)
+@click.option('-v', '--verbose', is_flag=True, default=False)
 @click.argument('block_schema_id')
 @click.argument('json_file')
 def block_schema_create_version(verbose, block_schema_id, json_file):
-    """Assign a json-schema file conforming to the EUDAT block-schema part of 
-    the root-schema that defines metadata format for this block schema. Upon 
-    success, this creates a new version, the old versions are maintained and 
+    """Assign a json-schema file conforming to the EUDAT block-schema part of
+    the root-schema that defines metadata format for this block schema. Upon
+    success, this creates a new version, the old versions are maintained and
     can be obtained by block_schema_list_versions subcommand."""
     try:
-        val = UUID(block_schema_id, version=4)
+        UUID(block_schema_id, version=4)
     except ValueError:
-        raise click.BadParameter("""BLOCK_SCHEMA_ID is not a valid UUID 
-        (hexadecimal numbers and dashes e.g. 
+        raise click.BadParameter("""BLOCK_SCHEMA_ID is not a valid UUID
+        (hexadecimal numbers and dashes e.g.
         fa52bec3-a847-4602-8af5-b8d41a5215bc )""")
+
     try:
         block_schema = BlockSchema.get_block_schema(schema_id=block_schema_id)
     except BlockSchemaDoesNotExistError:
         raise click.BadParameter("No block_schema with id %s" % block_schema_id)
+
     if not os.path.isfile(json_file):
-        raise click.ClickException("%s does not exist on the filesystem" % 
-        json_file)
-    f = open(json_file,'r')
+        raise click.ClickException("%s does not exist on the filesystem" %
+                                   json_file)
+
     schema_dict = {}
-    try:
-        schema_dict = json.load(f)
-    except json.decoder.JSONDecodeError:
-        raise click.ClickException("%s is not valid JSON" % json_file)
+    with open(json_file, 'r') as f:
+        try:
+            schema_dict = json.load(f)
+        except ValueError: # json decode error is subclass of ValueError
+            raise click.ClickException("%s is not valid JSON" % json_file)
+
     try:
         validate_metadata_schema(schema_dict)
```

Comment:
Ok I missread. I thought it was a call to `helpers.validate_json_schema`. It might be better to move this call to the API. For now it works because only the CLI can create schemas but this won't always be the case.